### ISSUE TITLE
Provide types for models

### ISSUE THREAD

**Issue by martynassateika** [martynassateika/CodeIgniter-phpstorm-plugin#15]
Version 0.4.0 introduced initial support for CodeIgniter 3's Models with a reference contributor and completion provider for model file names in `$this->load->model` (https://github.com/martynassateika/CodeIgniter-phpstorm-plugin/issues/6).

However even with this in place, loading model `Foo` using `$this->load->model('foo')` does not help PhpStorm.

* Set up an extension of `PhpTypeProvider2` that looks for `$this->XXX` field references.
* Check if XXX is a model.
* If so, set the type of the field reference to `XXX_model`.

**Comment by martynassateika** [comment]
Note that this won't work if people use the two-parameter `$this->load->model` version: https://www.codeigniter.com/user_guide/general/models.html#loading-a-model.

**Comment by GerB** [comment]
Hello,

Thank you for this plugin. It's a good start for me using PhpStorm instead of Netbeans, stuff like this is a huge time saver.
If you can get around providing full library and model support (including references), that would really be great. If I can be some kind of help with that (no Java expert, sorry), just let me know.

Thanks again.

**Comment by martynassateika** [comment]
Hey GerB, thanks a lot for the kind words. Happy to hear the plugin is useful.

I have started work on model support some time ago, but then life got in the way. I will revisit the codebase this weekend.

By the way, I have also recently set up a personal website which at the moment acts as a blog, but will also house information about my IntelliJ IDEA / PhpStorm projects. Currently the only way to figure out what the plugin _does_ is to look at the changelog. The site will expand more on each feature, kind of like a "What's New" page JetBrains put up whenever PhpStorm gets updated.
